packages/astro/src/core/app/index.ts

```diff
@@ -40,59 +40,78 @@ export interface MatchOptions {
 }
 
 export class App {
-	#env: Environment;
+	/**
+	 * The current environment of the application
+	 * @protected
+	 */
+	protected env: Environment;
 	#manifest: SSRManifest;
-	#manifestData: ManifestData;
+	protected manifestData: ManifestData;
 	#routeDataToRouteInfo: Map<RouteData, RouteInfo>;
-	#encoder = new TextEncoder();
-	#logging: LogOptions = {
+	protected encoder = new TextEncoder();
+	protected logging: LogOptions = {
```

Comment:
New classes that will extend this class will have access to this data; that's why I marked them `protected`.

Reply:
Are there existing super-classes that uses these fields? Or will you be creating those as a followup?

We might want to be cautious a bit as JS doesn't have a `protected` concept, and they will now be public with this change ([TypeScript example](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQMQAsCmIISAvNAORnqZYDu8ATiACbSkAMVGADvfAC55gAlgDNErcpUzQe-QcOgAjMPQkVUAXyA)). Not blocking this change though if you do plan to use this later.

Reply:
Yeah, maybe it's a bit premature this change. I am going to revert it.